internal/store/engine_state.go

```diff
@@ -82,6 +82,7 @@ type EngineState struct {
 	WatchSettings model.WatchSettings
 
 	PendingConfigFileChanges map[string]time.Time
+	PendingTiltfileTrigger   time.Time
```

Comment:
can you comment on why the tiltfile needs a trigger time, but other resources don't need a trigger time?

AFAICT, it seems like one trigger system uses a queue to control dupes, and the other trigger system uses time to control dupes. Not sure it's a good idea to use both, because then any future person editing this code has to learn the gotchas/subtleties of both approaches (as my comment above illustrates)

Reply:
I did it this way cuz the mechanisms for tiltfile updates and other manifest updates are so different--like it felt weird to me to use the queue for tiltfile builds when the tiltfile's position in the queue didn't affect ordering at all vis a vis everything else in the queue? but let me take a look and see how awkward it'll be in real life.